docs/developer-guide/plugins-scenarios.md

```diff
@@ -233,115 +231,79 @@ However there are some pitfalls you need to know:
-                .flatMap(buf -> {
-                    String content = buf.toString(UTF_8);
-                    buf.release();
-                    return sendToRemoteServer(content)
-                            .map(value -> copiedBuffer(value, UTF_8));
-                });
+The body transformation involves two lambda methods:
 
-        return chain.proceed(
-                request.newBuilder()
-                        .body(contentMapping)
-                        .build()
-        );
-    }
-    
-    StyxObservable<String> sendToRemoteServer(String buf) {
-        return StyxObservable.of("modified 3rd party content");
-    }
-}    
-```
+  1. A `.body(Function<ByteStream, ByteStream>)` accepts a lambda that modifies the request
+     byte stream. Here we provide a lambda that accepts a `ByteStream` and returns another
+     by applying a synchronous `map` operator on the stream.
+     
+  2. The `ByteStream` is modified by applying a `map` operator that synchronously modifies
+     each `Buffer` that streams through. The `map` involves another lambda that accepts a
+     `Buffer` and returns a modified buffer.
+  
+  3. In this example the buffer is converted to an upper case. 
+  
 
 ### Synchronously transforming streaming response content
 
+This example demonstrates how an HTTP response content can be transformed synchronously.
+
 ```java
 public class AsyncResponseContentStreamTransformation implements Plugin {
     @Override
-    public StyxObservable<HttpResponse> intercept(HttpRequest request, Chain chain) {
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
         return chain.proceed(request)
-                .map(response -> {
-                        StyxObservable<ByteBuf> contentMapping = response.body()
-                                .map(buf -> {
-                                    buf.release();
-                                    return copiedBuffer(buf.toString(UTF_8).toUpperCase(), UTF_8);
-                                });
-                        return response.newBuilder()
-                                .body(contentMapping)
-                                .build();
-                });
+            .map(response ->                                                                        // 1
+                response.newBuilder()                         
+                    .body(byteStream ->                                                             // 2
+                            byteStream.map(buf -> {                                                 // 3
+                                String upperCase = new String(buf.content(), UTF_8).toUpperCase();  // 4
+                                return new Buffer(upperCase, UTF_8);
+                            }))
+                    .build());
     }
 }
 ```
 
-### Asynchronously transforming streaming response content
+This is quite similar to the request transformation but it involves an additional lambda
```

Comment:
"This is quite similar to the request transformation but it involves an additional lambda
expression to capture the HTTP response from `Eventual` envelope:"
->
"This is quite similar to the request transformation, but it involves an additional lambda
expression to capture the HTTP response from its `Eventual` envelope:"

Reply:
👍